pkg/loki/modules.go

```diff
@@ -150,15 +150,23 @@ func (t *Loki) stopDistributor() (err error) {
 	return nil
 }
 
-func (t *Loki) initQuerier() (err error) {
+func (t *Loki) initQuerier() error {
 	level.Debug(util.Logger).Log("msg", "initializing querier worker", "config", fmt.Sprintf("%+v", t.cfg.Worker))
-	t.worker, err = frontend.NewWorker(t.cfg.Worker, httpgrpc_server.NewServer(t.server.HTTPServer.Handler), util.Logger)
+	worker, err := frontend.NewWorker(t.cfg.Worker, httpgrpc_server.NewServer(t.server.HTTPServer.Handler), util.Logger)
 	if err != nil {
-		return
+		return err
```

Comment:
was this kind of a bug that we didn't return err before?

Reply:
No, we used named return value before so it was ok, I've changed it. I shouldn't have. Let me put it back to minimize number of changes.

Reply:
I just find using explicit return values (and not using named return values) nicer. But it's a personal style, not a bug.

Reply:
Ah, now I see... since I use `worker, err :=` now, I wasn't sure about shadowing `err`, so just went with the safe choice of avoiding shadowing by removing it from named return. I think I'll keep it this way, if you don't mind.

Reply:
Good call shadowing is the worst !

Reply:
agreed